stag-library-compiler/src/main/java/com/vimeo/stag/processor/utils/TypeUtils.java

```diff
@@ -552,4 +558,45 @@ public static Element getElementFromTypeMirror(@NotNull TypeMirror typeMirror) {
         return getUtils().asElement(typeMirror);
     }
 
+
+    public enum JsonAdapterType {
+        NONE,
+        TYPE_ADAPTER,
+        TYPE_ADAPTER_FACTORY,
+        JSON_SERIALIZER,
+        JSON_DESERIALIZER,
+        JSON_SERIALIZER_DESERIALIZER
+
+    }
+    /**
+     * Return the type of JsonAdapter {@link TypeMirror}
+     *
+     * @param type :TypeMirror type
+     * @return {@link JsonAdapterType}
+     */
+    public static JsonAdapterType getJsonAdapterType(@NotNull TypeMirror type) {
+        WildcardType WILDCARD_TYPE_NULL = sTypeUtils.getWildcardType(null, null);
+        TypeMirror[] typex = {WILDCARD_TYPE_NULL};
+        if(sTypeUtils.isSubtype(type, sTypeUtils.getDeclaredType(ElementUtils.getTypeElementFromQualifiedName(TypeAdapter.class.getName()), typex))) {
+            return JsonAdapterType.TYPE_ADAPTER;
+        } else if(sTypeUtils.isAssignable(type, ElementUtils.getTypeFromQualifiedName(TypeAdapterFactory.class.getName()))) {
+            return JsonAdapterType.TYPE_ADAPTER_FACTORY;
+        } else {
+            boolean isDeserializer = sTypeUtils.isSubtype(type, sTypeUtils.getDeclaredType(ElementUtils.getTypeElementFromQualifiedName(JsonDeserializer.class.getName()), typex));
+            boolean isSerializer = sTypeUtils.isSubtype(type, sTypeUtils.getDeclaredType(ElementUtils.getTypeElementFromQualifiedName(JsonSerializer.class.getName()), typex));
```

Comment:
Since `sTypeUtils.getDeclaredType(ElementUtils.getTypeElementFromQualifiedName(JsonSerializer.class.getName())` is called twice in a row, could you break it out into its own variable? It would really improve readability here.